public/app/core/components/Page/usePageNav.ts

```diff
@@ -1,28 +1,38 @@
+import { useLocation } from 'react-router-dom';
 import { createSelector } from 'reselect';
 
 import { NavModel } from '@grafana/data';
 import { getNavModel } from 'app/core/selectors/navModel';
 import { store } from 'app/store/store';
 import { StoreState, useSelector } from 'app/types';
 
-export function usePageNav(navId?: string, oldProp?: NavModel): NavModel | undefined {
-  if (oldProp) {
-    return oldProp;
-  }
+import { getActiveItem } from '../NavBar/utils';
 
-  if (!navId) {
-    return;
+const selectNavModel = createSelector(
+  (state: StoreState) => state.navIndex,
+  (state: StoreState) => state.navBarTree,
```

Comment:
it's really confusing me that we have two identical redux stores for navtree :(  wish we can combine these into one store reducer / state that nav both the tree and an index (could add an index based on path then also)

Reply:
Yeah, I absolutely agree, would be nice to have a single source of truth. 👍